Automattic/abacus
# Wizard: filter metrics by tag

Fixes https://github.com/Automattic/experimentation-platform/issues/869

Renders tags as clickable Chip components to enable filtering in the Metrics step of the experiment wizard.
* Show selected (primary color) state for the active tag.
* Render loading state while fetching all tags.
* Use an "All" Chip as default state and to reset filtering.
* Only one tag can be applied at a time.
* Available only in debug mode.

**Demo:**

<img width="800" alt="Screenshot 2023-04-04 at 02 38 22" src="https://user-images.githubusercontent.com/14192054/229651045-a89adb34-4bf9-4981-9e2e-fde751275445.png">






## How has this been tested?






- Automated tests that cover added/changed functionality
- Manual testing with production data (locally)
- Manual testing with mock data (locally or on Vercel)
- Additional manual testing instructions (please specify):
  * Apply diff D104318-code and sandbox the wpcom API
  * Open the Metrics step of the Experiment Wizard and filter by metric tag
     * When "All" Chip tag is selected, all metrics should be displayed
     * The "assigned" metrics should remain regardless of the selected tag
